src/macaron/slsa_analyzer/git_service/api_client.py

```diff
@@ -264,6 +264,79 @@ def get_workflow_runs(self, full_name: str, branch_name: str, created_after: str
 
         return response_data
 
+    def get_workflow_run_jobs(self, full_name: str, run_id: str) -> dict:
+        """Query the GitHub REST API for the workflow run jobs.
+
+        The url would be in the following form:
+        ``https://api.github/com/repos/{full_name}/actions/runs/<run_id>/jobs``
+
+        Parameters
+        ----------
+        full_name : str
+            The full name of the target repo in the form ``owner/repo``.
+        run_id : str
+            The target workflow run ID.
+
+        Returns
+        -------
+        dict
+            The json query result or an empty dict if failed.
+
+        Examples
+        --------
+        The following call to this method will perform a query to
+        ``https://api.github/com/repos/{full_name}/
+        actions/runs/<run_id>/jobs``
+
+        >>> gh_client.get_workflow_run_jobs(
+            full_name="owner/repo",
+            run_id=<run_id>
+        )
+        """
+        logger.debug("Query GitHub to get run jobs for %s with run ID %s", full_name, run_id)
+
+        url = f"{GhAPIClient._REPO_END_POINT}/{full_name}/actions/runs/{run_id}/jobs"
+        response_data = send_get_http(url, self.headers)
+
+        return response_data
+
+    def get_workflow_run_for_date_time_range(self, full_name: str, datetime_range: str) -> dict:
```

Comment:
Instead of providing the `datetime_range` as a string, should we provide the range as `datetime` objects (which must have the same timezone of UTC) and then these `datetime` objects will be converted into the string format for the f query?

Reply:
I'm not sure what you mean. GitHub API accepts specific query formats, such as : `<start-datetime>..<end-datetime>` or simply  `<datetime>`. It is the responsibility of the caller of this function i.e., `ci_service.workflow_run_in_date_time_range()` to construct the final query value.